src/langcheck/metrics/eval_clients/_anthropic.py

```diff
@@ -0,0 +1,189 @@
+        if anthropic_client:
+            self._client = anthropic_client
+        elif use_async:
+            self._client = AsyncAnthropic()
+        else:
+            self._client = Anthropic()
+
+        self._anthropic_args = anthropic_args or {}
+        self._use_async = use_async
+
+    def _call_api(self,
+                  prompts: Iterable[str | None],
+                  config: dict[str, Any],
+                  *,
+                  tqdm_description: str | None = None) -> list[Any]:
+        # A helper function to call the API with exception filter for alignment
+        # of exception handling with the async version.
+        def _call_api_with_exception_filter(model_input: dict[str, Any]) -> Any:
+            if model_input is None:
+                return None
+            try:
+                return self._client.messages.create(**model_input)
+            except Exception as e:
+                return e
+
+        model_inputs = [{
+            "messages": [{
+                "role": "user",
+                "content": prompt
+            }],
+            **config
+        } for prompt in prompts]
+
+        if self._use_async:
+            # A helper function to call the async API.
+            async def _call_async_api() -> list[Any]:
+                responses = await asyncio.gather(*map(
+                    lambda model_input: self._client.messages.create(
+                        **model_input), model_inputs),
+                                                 return_exceptions=True)
+                return responses
+
+            responses = asyncio.run(_call_async_api())
+        else:
+            responses = [
+                _call_api_with_exception_filter(model_input)
+                for model_input in tqdm_wrapper(model_inputs,
+                                                desc=tqdm_description)
+            ]
+
+        # Filter out exceptions and print them out.
+        for i, response in enumerate(responses):
+            if not isinstance(response, Exception):
+                continue
+            print('Anthropic failed to return an assessment corresponding to '
+                  f'{i}th prompt: {response}')
+            responses[i] = None
+        return responses
+
+    def get_text_responses(
+            self,
+            prompts: Iterable[str],
+            *,
+            tqdm_description: str | None = None) -> list[str | None]:
+        '''The function that gets resonses to the given prompt texts.
+        We use Anthropic's 'claude-3-haiku-20240307' model by default, but you
+        can configure it by passing the 'model' parameter in the anthropic_args.
+
+        Args:
+            prompts: The prompts you want to get the responses for.
+
+        Returns:
+            A list of responses to the prompts. The responses can be None if the
+            evaluation fails.
+        '''
+        config = {
+            "model": "claude-3-haiku-20240307",
+            "max_tokens": 1024,
```

Comment:
nit: I feel like it's fine to keep this at the default value?

Reply:
Actually you need to specify a value, otherwise you get the following error

> Anthropic failed to return an assessment corresponding to 0th prompt: Missing required arguments; Expected either ('max_tokens', 'messages' and 'model') or ('max_tokens', 'messages', 'model' and 'stream') arguments to be given


I updated the limit to 4096